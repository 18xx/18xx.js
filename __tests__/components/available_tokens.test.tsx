import { mount } from 'enzyme';
import { List } from 'immutable';
import * as React from 'react';
import * as renderer from 'react-test-renderer';

import Company from '../../src/company';

import AvailableTokens from '../../src/components/available_tokens';

describe('AvailableTokens', () => {
  const companies: List<Company> = List([
    Company.fromJson(
      'B&O',
      {
        home: 'i15',
        name: 'Baltimore and Ohio Railroad',
        primaryColor: '#0B00F7',
        secondaryColor: 'white',
        shorthand: 'B&O',
        textColor: 'black',
        tokens: 3,
      },
    )
  ]);

  it('renderes a list of tokens', () => {
    const subject: React.ReactElement<AvailableTokens> = (
      <AvailableTokens companies={companies} onClick={jest.fn()} />
    );

    expect(renderer.create(subject)).toMatchSnapshot();
  });

  describe('onClick', () => {
    const fn: any = jest.fn();
    const subject: React.ReactElement<AvailableTokens> = (
      <AvailableTokens companies={companies} onClick={fn} />
    );

    it('calls the onClick with the company clicked', () => {
      mount(subject).find('span').simulate('click');
      expect(fn.mock.calls.length).toEqual(1);
      expect(fn.mock.calls[0][0]).toEqual(companies.get(0));
    });
  });
});
