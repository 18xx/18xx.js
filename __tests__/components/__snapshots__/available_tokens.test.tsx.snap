// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvailableTokens renderes a list of tokens 1`] = `
<div
  id="tokenMenu"
>
  <span
    onClick={[Function]}
  >
    <svg
      className="token"
      height="40"
      onContextMenu={[Function]}
      opacity={1}
      viewBox="0 0 40 40"
      width="40"
    >
      <circle
        cx={20}
        cy={20}
        fill="#0B00F7"
        r={16}
      />
      <rect
        fill="white"
        height={12}
        rx={1.75}
        ry={12}
        width={32}
        x={4}
        y={14}
      />
      <text
        fill="black"
        fontSize={10}
        fontWeight="normal"
        stroke="none"
        textAnchor="middle"
        x={20}
        y={24}
      >
        B&O
      </text>
    </svg>
  </span>
</div>
`;
